ch2driver/pytpcc/runtime/loader.py

```diff
@@ -393,29 +414,48 @@ def generateCustomer(self, c_w_id, c_d_id, c_id, sinceDate, badCredit, doesRepli
         c_since = sinceDate
         c_tuple = c_tuple + [c_since]
 
-        c_category_num = self.randomGen.nprng.choice(range(1, 128), size=self.randomGen.nprng.integers(0,15), replace=False)
-        c_categories = []
-        for c in c_category_num:
-            c_categories.append("category_" + str(format(c, "03d")))
+        c_categories, c_categories_flat = self.generateCustomerCategories(c_id, c_d_id, c_w_id)
+
         c_tuple.append(c_categories)
 
         c_data = self.randomGen.astring(constants.MIN_C_DATA, constants.MAX_C_DATA)
         c_tuple.append(c_data)
 
-        return c_tuple
+        return c_tuple, c_addresses_flat, c_phones_flat, c_categories_flat
     ## DEF
 
     ## ==============================================
     ## generateOrder
     ## ==============================================
-    def generateOrder(self, o_w_id, o_d_id, o_id, o_c_id, o_ol_cnt, orderTime, newOrder):
+    def generateOrder(self, o_w_id, o_d_id, o_id, o_c_id, o_ol_cnt, orderDate, orderTime, newOrder):
         """Returns the generated o_ol_cnt value."""
         o_entry_d = orderTime
         o_carrier_id = constants.NULL_CARRIER_ID if newOrder else self.randomGen.number(constants.MIN_CARRIER_ID, constants.MAX_CARRIER_ID)
         o_all_local = constants.INITIAL_ALL_LOCAL
         o_tuple = [ o_id, o_c_id, o_d_id, o_w_id, o_carrier_id, o_ol_cnt, o_all_local, o_entry_d ]
         o_tuple.append(self.generateExtraFields(self.maxExtraFields))
-        return o_tuple
+
+        startOrderLineDayRange = 2
+        endOrderLineDayRange = 151
+        ol_amount_idx = 5
+        ol_tuples = [ ]
+        total_ol_amount = 0
+        for ol_number in range(0, o_ol_cnt):
+            startOrderLineDate = orderDate + timedelta(days=startOrderLineDayRange)
+            endOrderLineDate = orderDate + timedelta(days=endOrderLineDayRange)
```

Comment:
These can be defined outside of the loop

Reply:
Done.